backend/battles/views.py

```diff
@@ -39,10 +43,28 @@ def get_context_data(self, **kwargs):
         user = User.objects.get(id=self.request.user.id)
         battle = self.kwargs["pk"]
         if user.teams.filter(battle=battle).exists():
-            context["user_has_team"] = True
-
+            messages.info(
+                self.request,
+                "You've chosen your team for this battle. Check your email for results",
+            )
         return context
 
+    def form_valid(self, form):
+        self.object = form.save()
+        battle = self.object.battle
+        creator = battle.user_creator
+        opponent = battle.user_opponent
+
+        if self.object.trainer == battle.user_creator:
+            send_invite_to_match(
+                creator.email,
+                opponent.email,
+                self.request.build_absolute_uri(f"/create-team/{battle.id}"),
+            )
+        if self.object.trainer == battle.user_opponent:
+            run_battle_task.delay(battle.pk, self.request.build_absolute_uri("/"))
```

Comment:
Nice way to get the full path!

Reply:
Thanks, it was Amanda that told me about it!